provider/cluster_rosa_classic_resource.go

```diff
@@ -1257,6 +1257,22 @@ func (r *ClusterRosaClassicResource) Update(ctx context.Context, request tfsdk.U
 		clusterBuilder.DisableUserWorkloadMonitoring(plan.DisableWorkloadMonitoring.Value)
 	}
 
+	_, shouldPatchProperties := common.ShouldPatchMap(state.Properties, plan.Properties)
+	shouldPatchProperties = shouldPatchProperties || len(state.OCMProperties.Elems) != len(OCMProperties)
```

Comment:
This is wrong - `state.OCMProperties` includes `state.Properties`, so `len(state.OCMProperties.Elems) != len(OCMProperties)` might be true even when shouldn't patch